lightning/src/routing/router.rs

```diff
@@ -343,19 +348,36 @@ pub struct RouteParameters {
 
 	/// The amount in msats sent on the failed payment path.
 	pub final_value_msat: u64,
+}
 
-	/// The CLTV on the final hop of the failed payment path.
-	///
-	/// This field is deprecated, [`PaymentParameters::final_cltv_expiry_delta`] should be used
-	/// instead, if available.
-	pub final_cltv_expiry_delta: u32,
+impl Writeable for RouteParameters {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		write_tlv_fields!(writer, {
+			(0, self.payment_params, required),
+			(2, self.final_value_msat, required),
+			(4, self.payment_params.final_cltv_expiry_delta, required),
```

Comment:
Could we comment that this is for forwards compat with 0.0.114? I presume in ~a few years we could simplify `payment_params` serialization and always assume the field is present